src/docs.js

```diff
@@ -6,38 +6,48 @@ let path = require('path');
 let recursiveReadSync = require('recursive-readdir-sync');
 
 async function documenter(options) {
-
   options = _.defaults({}, options, {
+    metadata: null,
     schemas: [],
     docsFolder: null,
+    references: [],
   });
   assert(options.schemas, 'options.schemas must be given');
   assert(options.schemas instanceof Array, 'options.schemas must be an array');
 
-  let schemas = options.schemas;
   let tarball = tar.pack();
+  let metadata = options.metadata;
+
+  if (metadata) {
```

Comment:
ah,

In that case we should ensure that metadata properties like menuIndex and
tier have sane values.. Ie. a simple assert will do..
Always validate input as we will be giving it many places...

--
Regards Jonas Finnemann Jensen.

2016-07-20 10:58 GMT-07:00 Brian Stack <notifications@github.com>:

> In src/docs.js
> <https://github.com/taskcluster/taskcluster-lib-docs/pull/7#discussion_r71575082>
> :
>
> >    let tarball = tar.pack();
> > +  let metadata = options.metadata;
> > +
> > +  if (metadata) {
>
> metadata is just this stuff:
> https://public.etherpad-mozilla.org/p/jonasfj-bstack-kt-taskcluster-docs-remake#lineNumber=107
>
> —
> You are receiving this because you were assigned.
> Reply to this email directly, view it on GitHub
> <https://github.com/taskcluster/taskcluster-lib-docs/pull/7/files/711349dee3efe9d99d0868d117b38ead2853f2e4#r71575082>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAJI5L40PbETtCCYmRa8nLM712NfWcJ9ks5qXmHWgaJpZM4JOV19>
> .
>